client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -0,0 +1,549 @@
+/* eslint no-mixed-operators: "off" */
+/* eslint react/sort-comp: "off" */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Canvas } from 'react-fabricjs'; // eslint-disable-line no-unused-vars
+
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import { scribingTools, scribingShapes, scribingToolColor,
+         scribingToolThickness, scribingToolLineStyle } from '../../constants';
+
+import { scribingShape } from '../../propTypes';
+
+const propTypes = {
+  answerId: PropTypes.number.isRequired,
+  scribing: scribingShape,
+  setCanvasLoaded: PropTypes.func.isRequired,
+  updateScribingAnswer: PropTypes.func.isRequired,
+  updateScribingAnswerInLocal: PropTypes.func.isRequired,
+};
+
+const styles = {
+  cover: {
+    position: 'fixed',
+    top: '0px',
+    right: '0px',
+    bottom: '0px',
+    left: '0px',
+  },
+  canvas_div: {
+    alignItems: 'center',
+    margin: 'auto',
+  },
+  canvas: {
+    width: '100%',
+    border: '1px solid black',
+  },
+  toolbar: {
+    marginBottom: '1em',
+  },
+  custom_line: {
+    display: 'inline-block',
+    position: 'inherit',
+    width: '25px',
+    height: '21px',
+    marginLeft: '-2px',
+    transform: 'scale(1.0, 0.2) rotate(90deg) skewX(76deg)',
+  },
+  tool: {
+    position: 'relative',
+    display: 'inline-block',
+    paddingRight: '24px',
+  },
+};
+
+export default class ScribingCanvas extends React.Component {
+  constructor() {
+    super();
```

Comment:
I think we should follow the [recommendation](https://facebook.github.io/react/docs/state-and-lifecycle.html#adding-local-state-to-a-class) to do `constructor(props) { super(props) ... }`, although there seems to be [debate](https://stackoverflow.com/questions/30571875/whats-the-difference-between-super-and-superprops-in-react-when-using-e) as to the usefulness of this.